zenoh-flow-python/zenoh_flow/types/__init__.py

```diff
@@ -72,3 +75,196 @@ def __repr__(self):
 
     def __str__(self):
         return f"Timestamp(ntp={self.ntp}, id={self.id})"
+
+
+class DataMessage:
+    """
+    Zenoh Flow data messages
+    It contains the actual data, the timestamp associated, and
+    information whether the message is a `Watermark`
+    """
+    def __init__(self, data: Any, ts: int,  watermark: bool):
+        self.__data = data
+        self.__ts = ts
+        self.__watermark = watermark
```

Comment:
Instead of having a `self.__watermark` field in the class — which forces us to maintain a coherent state — can't we just check if `self.__data` is `None`?

Reply:
As it `self.__data` cannot become easily an option, and given that some bad user can send an empty array.
I will leave it as it is, anyway, this information comes from rust, so it is always coherent.